### PR TITLE
Add meaningful `reasons` lists and get rid of redundant error handlers in steps

### PR DESCRIPTION
**Pull request by radix**
Few related changes

- CreateServer's success-case RETRY now returns `RETRY, ['waiting for server to become active']`, so we can distinguing between RETRY due to error and RETRY due to success.
- got rid of a few `report_failure` errbacks that were just handling arbitrary errors that were written before I added that logic to steps_to_effect.
- add exception to results list in CreateServer

### CODE REVIEW

File: otter/util/http.py [hunk 1]
```diff
@@ -159,6 +161,8 @@ def _segments(segments):
     return uri
 
 
+@attributes(['code', 'body',
```
Comment (manishtomar) [inline otter/util/http.py]: Why is this needed now?

Comment (radix) [inline otter/util/http.py]: I added it so I could easily compare instances of APIError in the unit tests

**Comment by radix** [comment]
jenkins please run unit tests

File: otter/convergence/steps.py [hunk 2]
```diff
@@ -144,7 +144,7 @@ def report_failure(result):
                 if message is not None:
                     return StepResult.FAILURE, [message]
 
-            return StepResult.RETRY, []
+            return StepResult.RETRY, [result[1]]
```
Comment (manishtomar) [inline otter/convergence/steps.py]: Since this code happens to have extracted `result`, it will be better to use `error` instead of `result[1]` instead.

**Comment by manishtomar** [comment]
LGTM

**Comment by radix** [comment]
jenkins please run unit tests

**Comment by lvh** [comment]
jenkins please run integration tests

**Comment by radix** [comment]
merging, thanks :)
